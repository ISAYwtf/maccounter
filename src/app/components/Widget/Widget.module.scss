.container {
  margin: 0 var(--size-space-between-component) var(--size-space-between-component) 0;
}

.title {
  display: flex;
  align-items: center;

  p {
    padding-right: 20px;
  }

  svg {
    width: 15px;
    height: 15px;
    cursor: pointer;
  }
}

.card {
  position: relative;

  .cardTitle {}

  .cardNumber {
    position: absolute;
    bottom: calc(var(--size-font-side) * -1);
    color: var(--color-side);
    font-size: var(--size-font-side);
  }
}

.sum {}

.dots {
  width: fit-content;
  justify-content: center;
  cursor: pointer;
}
