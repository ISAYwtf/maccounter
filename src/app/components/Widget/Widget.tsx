import React from 'react'
import styles from './Widget.module.scss'

interface WidgetProps {
    type: string
}

const Widget: React.FC<WidgetProps> = ({ type }) => (
    <div className={`component ${styles.container}`}>
        <p>{type}</p>
    </div>
)

export default Widget
