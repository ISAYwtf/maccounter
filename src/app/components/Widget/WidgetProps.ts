import { BasePropsWithoutChild } from '@localTypes/BaseProps'
import { ExpensesInitialState } from '@store/expenses-store/expensesInitialState'

export interface WidgetProps extends BasePropsWithoutChild {
    title: string,
    expenses: ExpensesInitialState
}
