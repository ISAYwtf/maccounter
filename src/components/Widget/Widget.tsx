import React from "react"
import styles from "./Widget.module.scss"

interface WidgetProps {
    type: string
}

const Widget: React.FC<WidgetProps> = (props) =>
    <div className={`component ${styles.container}`}>
        <p>{props.type}</p>
    </div>

export default Widget
