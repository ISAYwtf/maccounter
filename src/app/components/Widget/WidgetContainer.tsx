import { useAppSelector } from '@store/hooks'
import React from 'react'
import Widget from '@components/Widget/Widget'
import { WidgetProps } from '@components/Widget/WidgetProps'

const WidgetContainer: React.FC<Pick<WidgetProps, 'title'>> = ({ title }) => {
    const expensesData = useAppSelector((state) => state.expenses)
    return (
        <Widget title={title} expenses={expensesData} />
    )
}

export default WidgetContainer
