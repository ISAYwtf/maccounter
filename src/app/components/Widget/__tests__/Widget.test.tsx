import React from 'react'
import { render } from '@testing-library/react'
import Widget from '@components/Widget/Widget'

describe('Widget', () => {
    const renderComponent = <Widget type="list" />

    it('render Widget', () => {
        const { asFragment } = render(renderComponent)
        expect(asFragment()).toMatchSnapshot()
    })
})
