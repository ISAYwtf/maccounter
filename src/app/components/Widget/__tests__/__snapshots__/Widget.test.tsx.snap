// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget render Widget 1`] = `
<DocumentFragment>
  <div
    class="component container"
  >
    <div
      class="txt--title-small title"
    >
      <p>
        Last expenses
      </p>
      <svg>
        add.svg
      </svg>
    </div>
    <div
      class="rows"
    >
      <div
        class="row"
      >
        <div
          class="column card"
        >
          <p
            class="cardTitle"
          >
            Mastercard
          </p>
          <p
            class="cardNumber"
          >
            ****9474
          </p>
        </div>
        <div
          class="column sum"
        >
          <p>
            3,000 ₽
          </p>
        </div>
        <div
          class="column dots"
        >
          <svg>
            dots.svg
          </svg>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column card"
        >
          <p
            class="cardTitle"
          >
            Mastercard
          </p>
          <p
            class="cardNumber"
          >
            ****9474
          </p>
        </div>
        <div
          class="column sum"
        >
          <p>
            3,000 ₽
          </p>
        </div>
        <div
          class="column dots"
        >
          <svg>
            dots.svg
          </svg>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column card"
        >
          <p
            class="cardTitle"
          >
            Mastercard
          </p>
          <p
            class="cardNumber"
          >
            ****9474
          </p>
        </div>
        <div
          class="column sum"
        >
          <p>
            17,000 ₽
          </p>
        </div>
        <div
          class="column dots"
        >
          <svg>
            dots.svg
          </svg>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column card"
        >
          <p
            class="cardTitle"
          >
            Mastercard
          </p>
          <p
            class="cardNumber"
          >
            ****9474
          </p>
        </div>
        <div
          class="column sum"
        >
          <p>
            1,150,000 ₽
          </p>
        </div>
        <div
          class="column dots"
        >
          <svg>
            dots.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
